ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -299,12 +303,46 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 			isJobTerminal = isJobTerminal && finished
 		}
 
+		// If in SidecarMode, refine terminal condition by checking the sidecar container terminated.
```

Comment:
You can check the logic in `checkK8sJobAndUpdateStatusIfNeeded`, and try to unify the logic for both K8sMode and SidecarMode.